crates/dojo-language-server/src/bin/language_server.rs

```diff
@@ -17,15 +18,22 @@ async fn main() {
     #[cfg(feature = "runtime-agnostic")]
     let (stdin, stdout) = (stdin.compat(), stdout.compat_write());
 
+    let metadata = read_metadata(None).unwrap_or_else(|error| {
+        panic!("Problem reading metadata: {error:?}");
+    });
+
     let mut plugins = get_default_plugins();
     plugins.push(Arc::new(DojoPlugin { world_config: WorldConfig::default() }));
```

Comment:
once we have a scarb compiler for dojo, I think we'll be able to replace the `WorldConfig` with target compiler params

Reply:
I think you would rather write something like this in Scarb.toml:

```toml
[package]
# ...

[tool.dojo.world]
# ...  WorldConfig
```

We need a mechanism for adding tool-specific metadata for Protostar, so we'll work on this on our side, I think this could land maybe in Scarb 0.2.0

https://github.com/software-mansion/scarb/issues/136